source/code/scxsystemlib/common/scxostypeinfo.cpp

```diff
@@ -52,7 +52,7 @@ using namespace SCXCoreLib;
 
 namespace {
 
-#if defined(linux)
+#if defined(linux) && defined(PF_DISTRO_SUSE)
```

Comment:
I think this is still wrong. This code is needed for Linux and ULINUX. Consider that, right now, this code works regardless (to the best of my knowledge), although we normally only build for ULINUX.

The intent of this is that it runs on all Linux platforms, regardless of if we're building for universal (i.e. generic SSL) or not.

Perhaps it would help if you told me exactly what it is you're trying to fix? This code is generic code, designed to run on any Linux platform. It needs to determine, at runtime (not at build time), if special things are needed (based on capability of the O/S, and not based on version of the O/S).